### PR TITLE
Support lyft/ratelimit-style log level value case

### LINKED ISSUE #82: "LOG_LEVEL=info" is not supported
We should support the same log level names lyft/ratelimit does. Currently we support "LOG_LEVEL=Info" but not "LOG_LEVEL=info".

### PR DESCRIPTION
**Pull request by mdimjasevic**
This PR adds support for log level value case as in `lyft/ratelimit`, e.g., "debug" instead of only "Debug" so far. This is the value of the `LOG_LEVEL` environment variable.

It closes #82.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (neongreen) [inline README.md]: Accidental?

File: lib/Fencer/Logger.hs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+-- | A module for adjusting log level values to the title case. This
+-- is needed to make sure tinylog works with value cases used by
+-- lfyt/ratelimit.
+module Fencer.Logger (adjustCase) where
```
Comment (neongreen) [inline lib/Fencer/Logger.hs]: This should be more principled.

lyft/ratelimit uses the logrus library for logging. It defines the following log levels:

https://github.com/sirupsen/logrus/blob/39a5ad12948d094ddd5d5a6a4a4281f453d77562/logrus.go#L25

We should use the following mapping to match logrus:

```haskell
"panic" -> Fatal -- tinylog does not have Panic
"fatal" -> Fatal
"error" -> Error
"warn" -> Warn
"warning" -> Warn
"info" -> Info
"debug" -> Debug
"trace" -> Trace
```

Next, overwriting an environment variable is not the best approach.

* I would suggest using `tinylog-0.15.0`, which provides an option called `readEnvironment`: https://hackage.haskell.org/package/tinylog-0.15.0/docs/System-Logger.html#v:readEnvironment. If you set it to False, tinylog won't read env variables by itself, so you can feed whatever you want to it.
* We can then get rid of `LOG_BUFFER` and friends (they are kinda obscure and increase the surface level of Fencer for no good reason).
* You can read the variable `LOG_LEVEL` manually, lowercase it (no need for `titlecase` anymore), and transform it to a `Level` with the snippet above.

Comment (neongreen) [reply lib/Fencer/Logger.hs]: And you can do the LOG_LEVEL parsing in `Fencer.Settings`, so `Fencer.Logger` becomes redundant.

File: README.md (hunk 1 above)
Comment (mdimjasevic) [reply README.md]: Right! Fixed it.

File: lib/Fencer/Logger.hs (hunk 2 above)
Comment (mdimjasevic) [reply lib/Fencer/Logger.hs]: Can you be more specific about what variables you had in mind in the following?

> We can then get rid of LOG_BUFFER and friends (they are kinda obscure and increase the surface level of Fencer for no good reason).

Comment (neongreen) [reply lib/Fencer/Logger.hs]: `LOG_BUFFER`, `LOG_NETSTR`, `LOG_LEVEL_MAP`.

File: lib/Fencer/Settings.hs [hunk 3]
```diff
@@ -34,6 +37,10 @@ data Settings = Settings
       -- | @GRPC_PORT@: gRPC port to run the server on. The default
       -- value is 'defaultGRPCPort'.
     , settingsGRPCPort :: Port
+      -- | @LOG_LEVEL@: the logging level. This is a value translated
```
Comment (neongreen) [inline lib/Fencer/Settings.hs:41]: This should also say the the default is `'Logger.Debug'`.

File: lib/Fencer/Settings.hs [hunk 4]
```diff
@@ -66,3 +79,21 @@ parseBool s
...
     | otherwise = Nothing
+
+-- | Parse the string value of the LOG_LEVEL environment variable and
+-- map it to a 'System.Logger.Level'. Use the default Debug level if
+-- the environment variable was not set.
+parseLogLevel :: Maybe String -> Logger.Level
+parseLogLevel Nothing  = Logger.Debug
+parseLogLevel (Just s) = case readMaybe @Logger.Level s of
+  Just l  -> l
+  Nothing -> case (toLower <$> s) of
```
Comment (neongreen) [inline lib/Fencer/Settings.hs]:
```suggestion
  Nothing -> case toLower <$> s of
```

**Review by neongreen — APPROVED** [review]

File: lib/Fencer/Settings.hs (hunk 3 above)
Comment (mdimjasevic) [reply lib/Fencer/Settings.hs:41]: Fixed in an upcoming commit.

File: lib/Fencer/Logger.hs (hunk 2 above)
Comment (mdimjasevic) [reply lib/Fencer/Logger.hs]: In the upcoming commit I removed these 3 `LOG_` variables that you suggested.

Comment (mdimjasevic) [reply lib/Fencer/Logger.hs]: I implemented all of the remarks from above.

**Review by neongreen — APPROVED** [review]

**Pull request merged**